src/target/target.cc

```diff
@@ -375,7 +403,7 @@ Target::Target(const Map<String, ObjectRef>& config) {
 
 Target::Target(Target target, Target host) {
   ObjectPtr<TargetNode> n = make_object<TargetNode>(*target.get());
-  CHECK(!n->host.defined())
+  CHECK(!n->host.defined() || n->host == host)
```

Comment:
Use pointer equality explicitly

```suggestion
  CHECK(!n->host.defined() || n->host.same_as(host))
```